pkg/ingester/client/custom_test.go

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package client
+
+import (
+	"testing"
+	"unsafe"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/pkg/mimirpb"
+)
+
+func TestTimeSeriesChunk_CloneUnsafe(t *testing.T) {
+	const (
+		origLabelName  = "name"
+		origLabelValue = "value"
+	)
+	labelNameBytes := []byte(origLabelName)
+	labelValueBytes := []byte(origLabelValue)
+	c := TimeSeriesChunk{
+		Labels: []mimirpb.LabelAdapter{
+			{
+				Name:  yoloString(labelNameBytes),
+				Value: yoloString(labelValueBytes),
+			},
+		},
+		Chunks: []Chunk{
+			{
+				Data: mimirpb.UnsafeByteSlice(labelValueBytes),
```

Comment:
This is a bit confusing to me - is there a reason to reuse `labelValueBytes` here? Should we use a different byte slice for the chunks?

Reply:
I re-used because I didn't see any strict need to have a separate reference for the chunk, but I've pushed a revision with a dedicated one.